doc/contribute/guidelines.rst

```diff
@@ -504,8 +588,102 @@ before pushing on zephyr repo. To do this, make the file
 If you want to override checkpatch verdict and push you branch despite reported
 issues, you can add option --no-verify to the git push command.
 
-A more complete alternative to this is using check_compliance.py script from
-ci-tools repo.
+A more complete alternative to this is using :ref:`check_compliance_py` script.
+
+clang-format
+------------
+
+The `clang-format tool <https://clang.llvm.org/docs/ClangFormat.html>`_ can
+be helpful to quickly reformat large amounts of new source code to our
+`Coding Style`_ standards together with the ``.clang-format`` configuration file
+provided in the repository. ``clang-format`` is well integrated into most
+editors, but you can also run it manually like this:
+
+.. code-block:: bash
+
+   clang-format -i my_source_file.c
+
+``clang-format`` is part of LLVM, which can be downloaded from the project
+`releases page <https://github.com/llvm/llvm-project/releases>`. Note that if
+you are a Linux user, ``clang-format`` will likely be available as a package in
+your distribution repositories.
+
+When there are differences between the `Coding Style`_ guidelines and the
+formatting generated by code formatting tools, the `Coding Style`_ guidelines
+take precedence. Maintainer preference takes precedence over both the guidelines
+and the formatting tools.
```

Comment:
I don't think it's a good idea to let maintainers override the official coding guidelines.   I realize there may be exceptions, but exceptions should be rare and handled on a case by case basis.  

The text here implies the maintainer can adopt any style they want - for instance using 4 spaces in place of hard tabs.

```suggestion
take precedence. 
```

Reply:
Ouch, yeah sorry I missed that -- yeah anything that is unambiguously defined in the coding style takes precedence obviously, the maintainer IMO should have priority on ambiguous details. Thanks Keith.

Reply:
You're right, the text was misleading. Please check if the wording is better now.